### ISSUE TITLE
Product creation workflow improvements

### ISSUE THREAD

**Issue by khalibloo** [mirumee/saleor#4178]
### Description
While creating a product, if I realize that the category/collection/product type that I want to use has not been created yet, there's no convenient way to do so (maybe through a popup). Instead, I have to discard the information I've already entered into the product and create the categories or product types first.

Also, if you have a category tree where some subcategories have identical names (like "Footwear" under "Men's fashion" and "Footwear" under "Ladies fashion"), there is no way to tell them apart in the dropdown when creating a new product. See screenshot below. Some breadcrumbs and/or tooltips showing the parent categories of each suggestion should solve this nicely.

### Screenshots or mockups
![saleor dash issue](https://user-images.githubusercontent.com/6402969/58522950-8c185c00-81d3-11e9-8cc4-a341865167d0.png)

**Comment by maarcingebala** [comment]
I agree - @pwgryglak we'll need to think of a design for this component to somehow represent the nested structure.

**Comment by pwgryglak** [comment]
Designs for category tree representation are ready.

**Comment by maarcingebala** [comment]
@khalibloo For now we'll want to improve the category selection to render breadcrumbs for every choice including at least the category name and it's immediate ancestors, e.g. "Men > Clothing > Tshirts" etc.

We were also thinking about how to allow creating new categories from the product creation view. We have some ideas, but I was wondering if it wouldn't be simpler to allow for creating products without assigning categories. Then you could easily assign products to a category from the category details view.

**Comment by khalibloo** [comment]
For the categories ancestors, that's great. Breadcrumbs would work well, I believe. The tricky bit is how to handle the overflow that is sure to happen. Could we have tooltips showing the full list of ancestors? The tooltips could be as simple as the html title attribute. Or if you have other ideas to handle the overflow besides tooltips...

With regards to creating new categories, I find being able to create categories in product to be easier on the user. Being able to assign products to a category from the category details view is a very good option too, and would nicely supplement the first method, but I wouldn't choose it over the first method.

Part of my work with Saleor involves training the dashboard staff. And the biggest hurdle we have is that there is a rigid order of doing things. First, you create product type, then you create this and then that. It is a very unforgiving flow as any missed step results in a dead end. If we could have some degree of flexibility in the flow, that would ease my task a lot. If one misses a step, they would be able to remedy it.

That said, I also understand that the category creation form might be too big for a popup. In this case, it would still be better to have just name and description, for example, than to have to leave it blank and assign it afterwards from the category page. Also note that it is not obvious to the user that some fields can be left blank. Naturally, we'd want to fill every field and if one stops us in our track, we'd normally cut our losses and go back to the missed step than to risk filling in the other fields and hitting an error on submission.

It would definitely be good to have both of those category creation options. This way, different users can devise their own workflows. But I wouldn't pick the second over the first.

**Comment by maarcingebala** [comment]
@khalibloo This is very helpful feedback, thanks for sharing! We'll take it into consideration.

> And the biggest hurdle we have is that there is a rigid order of doing things.

Do you refer to the old static dashboard or the new one? I know this was a problem in the old one, but in 2.0 we want to provide much better UX and optimize the common patterns that admin users use.
